src/foam/core/CompoundException.java

```diff
@@ -7,19 +7,31 @@
 package foam.core;
 
 import java.util.ArrayList;
+import foam.nanos.app.AppConfig;
+import foam.nanos.app.Mode;
 
 public class CompoundException
   extends RuntimeException
 {
-  private ArrayList exceptions_ = new ArrayList<RuntimeException>();
+  private ArrayList exceptions_ = new ArrayList<Throwable>();
 
   public void add(Throwable t) {
     exceptions_.add(t);
   }
 
-  public void maybeThrow() {
-    if ( exceptions_.size() != 0 ) {
-      throw this;
+  public String toString() {
+    return "CompoundException [exceptions_=" + exceptions_ + "]";
+  }
+
+  public String getMessage() {
+    return super.getMessage();
+  }
+
+  public void maybeThrow(X x) {
+    if ( Mode.TEST == ((AppConfig) x.get("appConfig")).getMode() ) {
+      if ( exceptions_.size() != 0 ) {
+        throw this;
```

Comment:
maybeThrow should always through when exceptions_.size() > 0. 
The test for TEST was for you generating a System.err  or printStackTrace that you had before.

Reply:
@kgrgreer ask to keep it

Reply:
@hchoura yes, keep the test and generate the standout call you need. but now this only rethrows in Test mode, that is incorrect.